controllers/storagecluster/reconcile.go

```diff
@@ -676,3 +699,61 @@ func validateOverprovisionControlSpec(sc *ocsv1.StorageCluster, reqLogger logr.L
 	}
 	return nil
 }
+
+func (r *StorageClusterReconciler) ensureToolsDeployment(sc *ocsv1.StorageCluster) error {
+
+	var isFound bool
+	namespace := sc.Namespace
+
+	tolerations := []corev1.Toleration{{
+		Key:      defaults.NodeTolerationKey,
+		Operator: corev1.TolerationOpEqual,
+		Value:    "true",
+		Effect:   corev1.TaintEffectNoSchedule,
+	}}
+
+	tolerations = append(tolerations, sc.Spec.ManagedResources.CephToolbox.Tolerations...)
+
+	toolsDeployment := sc.NewToolsDeployment(tolerations)
+	foundToolsDeployment := &appsv1.Deployment{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: rookCephToolDeploymentName, Namespace: namespace}, foundToolsDeployment)
+
+	if err == nil {
+		isFound = true
+	} else if errors.IsNotFound(err) {
+		isFound = false
+	} else {
+		return err
+	}
+
+	// Checking spec of ocsinitialization for its Enablecephtools field
+	ocsinit := &ocsv1.OCSInitialization{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: "ocsinit", Namespace: namespace}, ocsinit)
+	if err != nil {
+		return err
+	}
+
+	if sc.Spec.EnableCephTools || ocsinit.Spec.EnableCephTools {
+		// Create or Update if ceph tools is enabled.
+
+		//Adding Ownerreference to the ceph tools
+		err = controllerutil.SetOwnerReference(sc, toolsDeployment, r.Client.Scheme())
+		if err != nil {
+			return err
+		}
+
+		if !isFound {
+			return r.Client.Create(context.TODO(), toolsDeployment)
+		} else if !reflect.DeepEqual(foundToolsDeployment.Spec, toolsDeployment.Spec) {
+
+			updateDeployment := foundToolsDeployment.DeepCopy()
+			updateDeployment.Spec = *toolsDeployment.Spec.DeepCopy()
+
+			return r.Client.Update(context.TODO(), updateDeployment)
+		}
+	} else if isFound && !sc.Spec.EnableCephTools && !ocsinit.Spec.EnableCephTools {
```

Comment:
when evaluating this condition you already know that both  `sc.Spec.EnableCephTools` and `ocsinit.Spec.EnableCephTools` are false, because if any of them was true you will not get to evaluate the else condition.

```suggestion
	} else if isFound  {
```

Reply:
Done